### ISSUE TITLE
error: ‘HMAC_CTX_new’ was not declared in this scope

### ISSUE THREAD

**Issue by nathanielzacharias** [voidloop/krakenapi#5]
Hi there, after generating the makefile using cmake, I get the following error in Terminal when I run 'make' : 

Any insight into how I can fix the problem would be appreciated. 
I'm using Ubuntu 16.04.4 LTS, gcc compiler version 5.4.0

```
`
/home/nz/Desktop/krakenapi-master/kraken/kclient.cpp: In function ‘std::vector<unsigned char> Kraken::hmac_sha512(const std::vector<unsigned char>&, const std::vector<unsigned char>&)’:
/home/nz/Desktop/krakenapi-master/kraken/kclient.cpp:88:33: error: ‘HMAC_CTX_new’ was not declared in this scope
    HMAC_CTX *ctx = HMAC_CTX_new();
                                 ^
/home/nz/Desktop/krakenapi-master/kraken/kclient.cpp:95:21: error: ‘HMAC_CTX_free’ was not declared in this scope
    HMAC_CTX_free(ctx);
                     ^
CMakeFiles/kapi.dir/build.make:62: recipe for target 'CMakeFiles/kapi.dir/kraken/kclient.cpp.o' failed
make[2]: *** [CMakeFiles/kapi.dir/kraken/kclient.cpp.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/kapi.dir/all' failed
make[1]: *** [CMakeFiles/kapi.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
`
```

EDIT: my openssl and libcurl4-openssl-dev packages are already the newest versions

**Comment by nathanielzacharias** [comment]
The HMAC_CTX_new call is only supported on openssl version 1.1.x.x which was not on my computer. 

Solution:

1. Go to https://www.openssl.org/source/ and download the version with extension 1.1.x.x
2. Unzip the tar.gz file.
3. Open the extracted folder
4. Open Terminal and run `./config -Wl,--enable-new-dtags,-rpath,'$(LIBRPATH)' && sudo make && sudo make install && openssl version -a
`
The steps above work. 

I tried the steps purely in Terminal but it didn't install into /usr/bin/ so I couldn't run 'openssl' (last line of commands in step 4). So I tried downloading to the Downloads folder by clicking on the link from the site, and the steps above worked.
